### ISSUE TITLE
🐞 Bug

### ISSUE THREAD

**Issue by devin-fisher** [CircuitVerse/CircuitVerse#5083]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

As part of a instruction counter circuit I was creating, multiple JK flipflops seem to be tied to the timing of the clock even though I was trying to divide the clock.

I've created a small project the demonstrates the bug: [Bug Project](https://circuitverse.org/users/259230/projects/bug-f0c3e273-8dc0-4105-ab21-22f01d7ae7dc)

You can see that the second JK FlipFlop toggles every clock cycle even though the simulator shows that the clock pin is only high every other clock cycle.

### Steps To Reproduce

I'm not totally sure what actually trigger this but (please see the project link above) but generally:
1. Use two JK FlipFlops
2. Tie both JK FlipFlops J and K pins high
3. Clock passes through a buffer ( I used an OR gate in original circuit) to JK FlipFlop 1 clock input 
4. With an `And` gate, and the clock signal with `Q Inverse` from JK FlipFLop 1 and pass that to JK FlipFlop 2 clock input
5. Observe both JK FlipFlops will toggle every clock cycle but it seem that JK FlipFlop only get a high signal in its clock input every other clock cycle. 

### Expected Behavior

I would expect that a JK FlipFlop to only toggle when it receives a clock signal

### Screenshots

Again see my sample project that demonstrates the bug: [Bug Project](https://circuitverse.org/users/259230/projects/bug-f0c3e273-8dc0-4105-ab21-22f01d7ae7dc)

### Device Information [optional]

```markdown
- OS: Windows 11
- Browser: Chrome
- version: Version 128.0.6613.120 (Official Build) (64-bit)
```

### Are you working on this issue?

No

**Comment by keshavagr273** [comment]
@devin-fisher Hey could you pls assign me this issue ?

**Comment by Adarsh-Dhar** [comment]
In the expected behavior both the jk flipflop should change in every clock right ?

**Comment by devin-fisher** [comment]
Yes, but in this case, it seems, that the second flip flop flips even when it doesn’t get at clock signal.

**Comment by pnk-ltoc-892** [comment]
can i work on this?

**Comment by salmoneatenbybear** [comment]
@gr455 @092vk can you please check if this needs work still ?

**Comment by SAMREEN111-hash** [comment]
Hi, I can reproduce this and I’m starting to investigate the simulator logic for clock handling in JK flip-flops. I’ll post findings or a PR if I can isolate the cause.

**Comment by Me-Priyank** [comment]
Simulator now lives in cv-frontend-vue repo
